packages/zowe-explorer/src/trees/shared/SharedInit.ts

```diff
@@ -311,23 +331,28 @@ export class SharedInit {
             });
             watcher.onDidChange(async (uri: vscode.Uri) => {
                 ZoweLogger.info(vscode.l10n.t("Team config file updated."));
-                const newProfileContents = await vscode.workspace.fs.readFile(uri);
-                if (newProfileContents.toString() === Constants.SAVED_PROFILE_CONTENTS.toString()) {
+                const newProfileContents = Buffer.from(await vscode.workspace.fs.readFile(uri));
+                if (Constants.SAVED_PROFILE_CONTENTS.get(uri.fsPath)?.equals(newProfileContents)) {
                     return;
                 }
-                Constants.SAVED_PROFILE_CONTENTS = newProfileContents;
+                Constants.SAVED_PROFILE_CONTENTS.set(uri.fsPath, newProfileContents);
                 void SharedActions.refreshAll();
                 ZoweExplorerApiRegister.getInstance().onProfilesUpdateEmitter.fire(Validation.EventType.UPDATE);
             });
         });
 
         try {
-            const zoweWatcher = imperative.EventOperator.getWatcher().subscribeUser(imperative.ZoweUserEvents.ON_VAULT_CHANGED, async () => {
-                ZoweLogger.info(vscode.l10n.t("Changes in the credential vault detected, refreshing Zowe Explorer."));
-                await ProfilesUtils.readConfigFromDisk();
-                await SharedActions.refreshAll();
-                ZoweExplorerApiRegister.getInstance().onVaultUpdateEmitter.fire(Validation.EventType.UPDATE);
-            });
+            // Workaround to skip ON_VAULT_CHANGED events triggered by ZE and not by external app
+            // TODO: Remove this hack once https://github.com/zowe/zowe-cli/issues/2279 is implemented
+            SharedInit.originalEmitZoweEvent = (imperative.EventProcessor.prototype as any).emitZoweEvent;
+            (imperative.EventProcessor.prototype as any).emitZoweEvent = function (eventName: string): void {
+                SharedInit.emitZoweEventHook(this, eventName);
+            };
```

Comment:
Nice use of closure 🙏 

This wouldn't have worked with arrow functions (unless you do some extra binds) 😋

Reply:
Thanks to @traeok for this one 😋